src/broadcast/macos/macosmediaplayerservice.mm

```diff
@@ -0,0 +1,277 @@
+    }
+}
+
+} // anonymous namespace
+
+MacOSMediaPlayerService::MacOSMediaPlayerService(
+        PlayerManagerInterface& pPlayerManager) {
+    // Set up deck attributes to track and control playback state
+    for (unsigned i = 1; i <= pPlayerManager.numberOfDecks(); i++) {
+        DeckAttributes* attributes =
+                new DeckAttributes(i, pPlayerManager.getDeck(i));
+        m_deckAttributes.append(attributes);
+        connect(attributes,
+                &DeckAttributes::playChanged,
+                this,
+                &MacOSMediaPlayerService::slotPlayChanged);
+        connect(attributes,
+                &DeckAttributes::playPositionChanged,
+                this,
+                &MacOSMediaPlayerService::slotPlayPositionChanged);
+    }
+
+    // Register command handlers for controlling Mixxx's playback state
+    // externally (i.e. via the command center). At least one handler must be
+    // registered, otherwise the now playing info will not show up.
+    MPRemoteCommandCenter* center = [MPRemoteCommandCenter sharedCommandCenter];
+    [center.playCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(
+            MPRemoteCommandEvent* event) {
+        bool success = updatePlayState(true);
+        return commandHandlerStatusFor(success);
+    }];
+    [center.pauseCommand addTargetWithHandler:^MPRemoteCommandHandlerStatus(
+            MPRemoteCommandEvent* event) {
+        bool success = updatePlayState(false);
+        return commandHandlerStatusFor(success);
+    }];
+    [center.togglePlayPauseCommand
+            addTargetWithHandler:^MPRemoteCommandHandlerStatus(
+                    MPRemoteCommandEvent* event) {
+                bool success = togglePlayState();
+                return commandHandlerStatusFor(success);
+            }];
+    [center.changePlaybackPositionCommand
+            addTargetWithHandler:^MPRemoteCommandHandlerStatus(
+                    MPRemoteCommandEvent* event) {
+                auto positionEvent =
+                        (MPChangePlaybackPositionCommandEvent*)event;
+                bool success = updatePlayPosition(positionEvent.positionTime);
+                return commandHandlerStatusFor(success);
+            }];
+
+    // TODO: Hook up 'next track' to Auto DJ
+
+    // Write up player info so we get notified about track updates
+    connect(&PlayerInfo::instance(),
+            &PlayerInfo::currentPlayingTrackChanged,
+            this,
+            &MacOSMediaPlayerService::slotBroadcastCurrentTrack);
+
+    // Wire up cover art cache so loaded covers get passed to our slot
+    connect(CoverArtCache::instance(),
+            &CoverArtCache::coverFound,
+            this,
+            &MacOSMediaPlayerService::slotCoverFound);
+}
+
+MacOSMediaPlayerService::~MacOSMediaPlayerService() {
+    for (DeckAttributes* attributes : qAsConst(m_deckAttributes)) {
+        delete attributes;
+    }
+}
```

Comment:
Implementation note: This approach of managing memory manually is based on https://github.com/mixxxdj/mixxx/blob/50c981c88647b383db1b4b9264a1942b75c8edf0/src/broadcast/mpris/mprisplayer.cpp#L76-L80 from #3483, I haven't found a better way to do this though since `DeckAttributes` doesn't seem to provide an easy way to hook itself into the `QObject` graph.